v1/nix/pkgs/fetchPipMetadata/src/tests/test_evaluate_extras.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+from packaging.requirements import Requirement
+
+import lock_file_from_report as l
+
+
+@pytest.fixture
+def env():
+    return dict(python_version="3.10")
+
+
+def test_marker_match(env):
+    env = dict(python_version="3.10")
+    requirement = Requirement("requests; python_version == '3.10'")
+    assert l.evaluate_extras(requirement, None, env) == True
+
+
+def test_marker_mismatch(env):
+    env = dict(python_version="3.10")
+    requirement = Requirement("requests; python_version >= '3.11'")
```

Comment:
```suggestion
    requirement = Requirement("requests; python_version == '3.11'")
```

I'd recommend either that or `>= '3.10'` to make intent clearer.